lib/bolt/inventory/target.rb

```diff
@@ -153,9 +154,24 @@ def validate
           raise Bolt::UnknownTransportError.new(transport, uri)
         end
 
+        validate_fact_names(@facts)
+
         transport_config
       end
 
+      # Validate fact names and issue a deprecation warning if any fact names have a dot.
+      #
+      private def validate_fact_names(facts)
+        if (dotted = facts.keys.select { |name| name.include?('.') }).any?
```

Comment:
I think we need to also dig into hashes and look for hash keys with `.`

Reply:
Apparently this is not the case.

Reply:
Phew. I wasn't too sure at first, but the doc for this issue suggested that we had internal facts with dotted names that should stay that way.

Reply:
Are only top-level dots considered structured facts?

Reply:
Answer: yes, nested facts with dots in them aren't treated as structured facts with Facter 4. Do we still want to discourage people from using them in general though? I'm not sure if that will ever change.